File: src/code/js/_035.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
//二分查找
var searchInsert = function (nums, target) {
    var l = 0, r = nums.length - 1;
    while (l <= r) {
        var mid = Math.floor((l + r) / 2);
        if (nums[mid] === target) {
            return mid;
        }
        if (nums[mid] < target) {
            l = mid + 1;
        } else {
            r = mid - 1;
        }
    }
    return l;
};

File: src/code/java/_035.java
package code.java;

public class _035 {
    // 二分查找
    // 二分查找如果找不到 最终的l,r值将满足 nums[r]<target<nums[l]
    public int searchInsert(int[] nums, int target) {
        int l = 0, r = nums.length - 1;
        while (l <= r) {
            int mid = (l + r) / 2;
            if (nums[mid] == target) {
                return mid;
            }
            if (nums[mid] < target) {
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }
        return l;
    }
}
